ecs-cli/modules/cli/cluster/cluster_app.go

```diff
@@ -321,6 +332,27 @@ func createCluster(context *cli.Context, awsClients *AWSClients, commandConfig *
 	return cfnClient.WaitUntilCreateComplete(stackName)
 }
 
+func determineArchitecture(cfnParams *cloudformation.CfnStackParams) (string, error) {
```

Comment:
I split this out into a separate function so that I can write unit tests for it; very open to suggestions for a better name for it tho.

Reply:
Hmm, name most consistent with this sort of thing I can think of is just `getArchitecture`, but I have no strong objections to `determineArchitecture`.